src/network/peer.rs

```diff
@@ -479,8 +480,16 @@ impl<N: Network, E: Environment> Peer<N, E> {
                                         continue;
                                     }
                                     // Retrieve the requested blocks.
-                                    let blocks = match ledger_reader.get_blocks(start_block_height, end_block_height) {
-                                        Ok(blocks) => blocks,
+                                    let blocks: Vec<Block<N>> = match ledger_reader.get_blocks(start_block_height, end_block_height) {
```

Comment:
```suggestion
                                    let blocks: Vec<Block<N>> = match ledger_reader.get_blocks(start_block_height, end_block_height).and_then(|blocks| blocks.collect()) {
```

Reply:
This is great! much cleaner now. The change was made in this commit [f4eccff](https://github.com/AleoHQ/snarkOS/pull/1753/commits/f4eccff4fbc5864b1ce3aeb3d519f1edf79bad88)